Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,38 +1,68 @@
 #include "BatteryStateOfChargeService.h"
+#include "BatteryData.h"
+
+#define HOURS_TO_MS (3600.0 * 1000.0)
 
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
 }
 
+static double prevCurrent, current, voltage, totalAmpHours;
+static QTime prevTime, currentTime;
+
+//static QTime timeUntilCharged, timeUntilDepleted;
+
+//Constructors
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+    initialAmpHours_ = BATTERY_AMP_HOUR_CAPACITY * (initialStateOfChargePercent/100);
+    prevTime = QTime(0,0);
+    totalAmpHours = initialAmpHours_;
+
 }
 
-BatteryStateOfChargeService::~BatteryStateOfChargeService()
+double BatteryStateOfChargeService::totalAmpHoursUsed() const //using the most recent data points
 {
-}
 
-double BatteryStateOfChargeService::totalAmpHoursUsed() const
+    totalAmpHours += ((prevCurrent + current)/2) * ((prevTime.msecsTo(currentTime)) / HOURS_TO_MS);
+
+    return totalAmpHours;
+}
+BatteryStateOfChargeService::~BatteryStateOfChargeService() //destructor
```

Comment:
In the future, re-place your code in the same line so that the reviewer doesn't accidentally think you removed it!